src/Symfony/Component/Cache/Adapter/PhpFilesAdapter.php

```diff
@@ -0,0 +1,227 @@
+        $ok = $this->filesCacheHelper->saveFile($file, $fileContent, $currentSequenceNumber);
+        if (function_exists('opcache_invalidate')) {
+            @opcache_invalidate($file, true);
+        }
+        if (function_exists('apc_compile_file')) {
+            @apc_compile_file($file);
+        }
+
+        return $ok;
+    }
+
+    /**
+     * @param string $file
+     *
+     * @return bool
+     */
+    private function removeCacheFile($file)
+    {
+        if (defined('HHVM_VERSION')) {
+            // workaround for https://github.com/facebook/hhvm/issues/1447
+            // save file with empty expired data instead of removing it
+            // so that we can check modification time (where we store version number)
+            return $this->saveCacheFile($file, null, 0);
+        }
+        if (function_exists('opcache_invalidate')) {
+            @opcache_invalidate($file, true);
+        }
+        if (function_exists('apc_delete_file')) {
+            @apc_delete_file($file);
+        }
+        @unlink($file);
+
+        return !@file_exists($file);
+    }
+
+    /**
+     * @param string $file File path
+     *
+     * @return mixed|null
+     */
+    private function includeCacheFileWithSequenceCheck($file)
+    {
+        $valueArray = $this->includeCacheFile($file);
+
+        if (!is_array($valueArray) || 3 !== count($valueArray)) {
+            return;
+        }
+
+        list(, , $foundSequenceNumber) = $valueArray;
+
+        if (defined('HHVM_VERSION')) {
+            // workaround for https://github.com/facebook/hhvm/issues/1447 and similar
+            // use file modification as a sequence number (cache file version)
+            @clearstatcache(true, $file);
+            $actualSequenceNumber = (int) @filemtime($file);
+            if ($foundSequenceNumber != $actualSequenceNumber) {
+                return $this->includeCacheFile($file, true);
+            }
+        }
+
+        return $valueArray;
+    }
+
+    /**
+     * @param string $file    File path
+     * @param bool   $useEval If true, tries to eval file contents instead of including it (forcing omitting opcache).
+     *
+     * @return mixed|null
+     */
+    private function includeCacheFile($file, $useEval = false)
+    {
+        if ($useEval) {
+            $content = @file_get_contents($file);
+            if ($content) {
+                $valueArray = eval('?>'.$content);
+            } else {
+                return;
+            }
+        } else {
+            $valueArray = @include $file;
+        }
+
+        return $valueArray;
+    }
+
+    /**
+     * @param mixed $value
+     * @param int   $expiresAt
+     * @param int   $currentSequenceNumber
+     *
+     * @return string
+     */
+    private function createCacheFileContent($value, $expiresAt, $currentSequenceNumber)
+    {
+        $exportedValue = var_export(serialize(array($value, $expiresAt, $currentSequenceNumber)), true);
```

Comment:
$expiresAt and $currentSequenceNumber should not be serialized, and $value should be unserialized only if expiration/seq number are fine.